cmd/argocd-util/commands/config.go

```diff
@@ -0,0 +1,193 @@
+	argocd-util config app jsonnet-guestbook --repo https://github.com/argoproj/argocd-example-apps.git --path jsonnet-guestbook --dest-namespace default --dest-server https://kubernetes.default.svc --jsonnet-ext-str replicas=2
+
+	# Generate declarative config for a Helm app
+	argocd-util config app helm-guestbook --repo https://github.com/argoproj/argocd-example-apps.git --path helm-guestbook --dest-namespace default --dest-server https://kubernetes.default.svc --helm-set replicaCount=2
+
+	# Generate declarative config for a Helm app from a Helm repo
+	argocd-util config app nginx-ingress --repo https://kubernetes-charts.storage.googleapis.com --helm-chart nginx-ingress --revision 1.24.3 --dest-namespace default --dest-server https://kubernetes.default.svc
+
+	# Generate declarative config for a Kustomize app
+	argocd-util config app kustomize-guestbook --repo https://github.com/argoproj/argocd-example-apps.git --path kustomize-guestbook --dest-namespace default --dest-server https://kubernetes.default.svc --kustomize-image gcr.io/heptio-images/ks-guestbook-demo:0.1
+
+	# Generate declarative config for a app using a custom tool:
+	argocd-util config app ksane --repo https://github.com/argoproj/argocd-example-apps.git --path plugins/kasane --dest-namespace default --dest-server https://kubernetes.default.svc --config-management-plugin kasane
+`,
+		Run: func(c *cobra.Command, args []string) {
+			var app v1alpha1.Application
+			if fileURL == "-" {
+				// read stdin
+				err := cmdutil.ReadAppFromStdin(&app)
+				errors.CheckError(err)
+			} else if fileURL != "" {
+				// read uri
+				err := cmdutil.ReadAppFromURI(fileURL, &app)
+				errors.CheckError(err)
+
+				if len(args) == 1 && args[0] != app.Name {
+					log.Fatalf("app name '%s' does not match app spec metadata.name '%s'", args[0], app.Name)
+				}
+				if appName != "" && appName != app.Name {
+					app.Name = appName
+				}
+				if app.Name == "" {
+					log.Fatalf("app.Name is empty. --name argument can be used to provide app.Name")
+				}
+				cmdutil.SetAppSpecOptions(c.Flags(), &app.Spec, &appOpts)
+				cmdutil.SetParameterOverrides(&app, appOpts.Parameters)
+				cmdutil.SetLabels(&app, labels)
+			} else {
+				// read arguments
+				if len(args) == 1 {
+					if appName != "" && appName != args[0] {
+						log.Fatalf("--name argument '%s' does not match app name %s", appName, args[0])
+					}
+					appName = args[0]
+				}
+				app = v1alpha1.Application{
+					TypeMeta: v1.TypeMeta{
+						Kind:       application.ApplicationKind,
+						APIVersion: application.Group + "/v1aplha1",
+					},
+					ObjectMeta: v1.ObjectMeta{
+						Name: appName,
+					},
+				}
+				cmdutil.SetAppSpecOptions(c.Flags(), &app.Spec, &appOpts)
+				cmdutil.SetParameterOverrides(&app, appOpts.Parameters)
+				cmdutil.SetLabels(&app, labels)
+			}
+			if app.Name == "" {
+				c.HelpFunc()(c, args)
+				os.Exit(1)
+			}
+
+			appClientset := appfake.NewSimpleClientset(&app)
```

Comment:
Why is a fake client required here? Unless I'm missing something you can just print app object?